pyspecProcScripts/first_level/QESR_rescale.py

```diff
@@ -0,0 +1,95 @@
+            assert type(calibration_name) is str, "the calibration name must be specified as a string!"
+            assert len(calibration_name)>0, "you MUST use a calibration name and set the values `[calibration_name] q` and `[calibration_name] propFactor` in your _pyspecdata file!"
+            self.current_calib_name = calibration_name
+            # note that this will fail if you don't have "[calibration_name] propFactor" and "[calibration_name] q" set in your pyspecdata file
+            try:
+                self.default_Q = float(pyspec_config.get_setting(f"{calibration_name} Q"))
+                self.dint_propFactor = float(pyspec_config.get_setting(f"{calibration_name} propFactor"))
+            except:
+                raise RuntimeError(f"I expect a line in the [General] block of your pyspecdata config file (in your home directory) that sets the value of the following variables:\n{calibration_name} q\n{calibration_name} propFactor\n\n(I can't run without these)")
+    def use_diameter(self, diameter_name):
+        if diameter_name != self.current_diam_name:
+            assert type(diameter_name) is str, "the diameter name must be specified as a string!"
+            assert len(diameter_name)>0, "you MUST use a diameter name and set the values `[diameter_name] q` and `[diameter_name] propFactor` in your _pyspecdata file!"
+            self.current_diam_name = diameter_name
+            try:
+                self.d = float(pyspec_config.get_setting(f"{diameter_name} diameter"))
+            except:
+                raise RuntimeError(f"(note this the second in a similar pair of errors) I expect a line in the [General] block of your pyspecdata config file (in your home directory) that sets the value of the following variables:\n{diameter_name} diameter")
+calibcache = calib_info()
+ureg = UnitRegistry(
+    system="mks", autoconvert_offset_to_baseunit=True, auto_reduce_dimensions=True
+)
+Q_ = ureg.Quantity
+def QESR_scalefactor(d, calibration_name=None, diameter_name=None):
+    """
+    Parameters
+    ==========
+    calibration_name:   str
+                        The key corresponding to the appropriate proportionality constant
+                        in your pyspecdata config file.
+                        Typically this is one value that doesn't need changing
+    diameter_name:      str
+                        The key corresponding to the diameter of the capillary tube
+                        used in the ESR experiment. This enables us to calculate a 
+                        reliable concentration regardless of which capillary was used.
+    Returns
+    =======
+    Denominatory of equation 2-17 divided by the proportionality constant. Also includes a 
+    factor of 1e-6 to convert Molar to micromolar.
+    """
+    calibcache.use_calibration(calibration_name)
+    calibcache.use_diameter(diameter_name)
+    # {{{ determine the signal denominator from the parameters of interest
+    G_R = Q_(*d.get_prop("Gain"))
```

Comment:
Why is all of this new?

Reply:
This is the point of the PR? To have a function for the rescaling on first level that the actual quantification will call

Reply:
My question is that I thought some amount of this code would have already been present... I thought this was more focused on just adding the diameter feature?

Reply:
no this is not in master